tests/test_utils/test_transactiontestcase.py

```diff
@@ -1,10 +1,48 @@
+import json
 from unittest import mock
 
+from django.apps import apps
 from django.db import connections
 from django.test import TestCase, TransactionTestCase, override_settings
 
+from .models import Car
 
-class TestSerializedRollbackInhibitsPostMigrate(TransactionTestCase):
+
+class TestSerializedContentMockMixin(object):
+    """
+    This mixin should be used on each test involving
+    TransactionTestCase and serialized_rollback = True option
+    to avoid test dependencies.
+    It mocks what would be serialized by Django after initial data migrations,
+    and restores it at the end of the test.
+    """
+    initial_data_migration = '[]'
+    _connections_test_serialized_content = {}
+
+    def _pre_setup(self):
+        for db_name in self._databases_names(include_mirrors=False):
+            self._connections_test_serialized_content[db_name] = connections[db_name]._test_serialized_contents
+            connections[db_name]._test_serialized_contents = self.initial_data_migration
+        super(TestSerializedContentMockMixin, self)._pre_setup()
+
+    def _post_teardown(self):
+        super(TestSerializedContentMockMixin, self)._post_teardown()
+        for db_name in self._databases_names(include_mirrors=False):
+            connections[db_name]._test_serialized_contents = self._connections_test_serialized_content[db_name]
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestSerializedContentMockMixin, cls).tearDownClass()
+
+        # Cleaning any data that has been created by our mixin class
+        # to not impact any other django suite test.
+        json_data = json.loads(cls.initial_data_migration)
+        for data in json_data:
```

Comment:
`for data in json.loads(...):`

Reply:
Also, can `initial_data_migration` be something other than JSON? Should this be using `django.core.serializers`?

Reply:
The data has been generated by a call to:serialize_db_to_string(self):
        """
        Serialize all data in the database into a JSON string.
        Designed only for test runner usage; will not handle large
        amounts of data.
        """